flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/aggfunctions/SortAggFunction.scala

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.table.functions.aggfunctions
+
+import java.math.BigDecimal
+import java.util.{List => JList}
+
+import org.apache.flink.api.common.typeinfo.{BasicTypeInfo, TypeInformation}
+import org.apache.flink.api.java.tuple.{Tuple2 => JTuple2}
+import org.apache.flink.api.java.typeutils.TupleTypeInfo
+import org.apache.flink.table.functions.{Accumulator, AggregateFunction}
+import scala.collection.mutable.ArrayBuffer
+import org.apache.flink.types.Row
+import org.apache.flink.table.runtime.aggregate.UntypedOrdering
+import org.apache.flink.table.runtime.aggregate.MultiOutputAggregateFunction
+import java.util.{ List => JList,ArrayList }
+import org.apache.flink.api.java.typeutils.RowTypeInfo
+import org.apache.flink.api.java.typeutils.ListTypeInfo
+import org.apache.calcite.rel.RelFieldCollation.Direction
+
+/** The initial accumulator for Sort aggregate function */
+class SortAccumulator extends ArrayBuffer[JTuple2[Row,Row]] with Accumulator with Serializable
+
+/**
+  * Base class for built-in Min aggregate function
+  *
+  * @tparam K the type for the key sort type
+  * @tparam T the type for the aggregation result
+  */
+abstract class SortAggFunction[K,T](
```

Comment:
@fhueske 
Both are fine - i do not think there is one particular simpler than the other. The reason i chose the aggregate-based implementation was based on the initial discussions to treat this as an aggregation as well. However, i can make another version in which i would move this in the processFunction.